components/chart/index.tsx

```diff
@@ -93,27 +95,19 @@ function Chart({ data }: { config: Config; data: GraphContent[] }) {
 
   const focusModule = useCallback(
     (id: string) => {
-      if (currentVersion && currentVersion.nodes) {
-        // const moduleNumberTrimmed = name.substring(0, name.indexOf('>')).substring(5);
-        const moduleId = parseInt(id);
-
-        Object.keys(currentVersion.nodes).forEach((key) => {
-          const node = currentVersion.nodes[key];
-          if (node.id === moduleId) {
-            const toX = node.x;
-            const toy = node.y;
-            const canvas = document.getElementById('canvas');
-            canvas.scrollLeft = toX - 500 > 0 ? toX - 500 : 0;
-            canvas.scrollTop = toy - 250 > 0 ? toy - 250 : 0;
-            selectModule(node);
-            setShowSearch(false);
-            setSearch({ value: '', isValid: true, message: '' });
-          }
-        });
-      }
+        if (currentVersion && currentVersion.nodes) {
+            const moduleId = parseInt(id);
+            const node = findModuleById(moduleId);
+            if (node) {
+                setSelectedModule(node);
+                setSelectedModules([moduleId]);
```

Comment:
```suggestion
selectModule(node);
```
You could use this one instead :)
However I did that no need to do anything, just letting you know about it